infra/src/test/kotlin/fi/fta/geoviite/infra/publication/PublicationServiceIT.kt

```diff
@@ -1911,6 +1941,144 @@ class PublicationServiceIT @Autowired constructor(
+                .copy(
+                    topologyStartSwitch = TopologyLocationTrackSwitch(switchAddedToTopologyStart.id, JointNumber(1)),
+                    topologyEndSwitch = TopologyLocationTrackSwitch(switchAddedToTopologyEnd.id, JointNumber(1))
+                ),
+            alignmentWithSwitchLinks(
+                switchAddedToAlignment.id,
+                switchMerelyRenamed.id,
+                newSwitchReplacingOldWithSameName.id,
+                null,
+            ),
+        )
+        publish(
+            publicationService,
+            testBranch,
+            locationTracks = listOf(originalLocationTrack.id),
+            switches = listOf(
+                switchDeleted.id,
+                switchMerelyRenamed.id,
+                originalSwitchReplacedWithNewSameName.id,
+                newSwitchReplacingOldWithSameName.id,
+            ),
+        )
+        locationTrackService.mergeToMainBranch(testBranch, originalLocationTrack.id)
+        listOf(
+            switchDeleted.id,
+            switchMerelyRenamed.id,
+            originalSwitchReplacedWithNewSameName.id,
+            newSwitchReplacingOldWithSameName.id
+        ).forEach { id -> switchService.mergeToMainBranch(testBranch, id) }
+        publish(
+            publicationService,
+            locationTracks = listOf(originalLocationTrack.id),
+            switches = listOf(
+                switchDeleted.id,
+                switchMerelyRenamed.id,
+                originalSwitchReplacedWithNewSameName.id,
+                newSwitchReplacingOldWithSameName.id,
+            ),
+        )
+        val latestPubs = publicationService.fetchLatestPublicationDetails(LayoutBranch.main, 2)
+        val latestPub = latestPubs[0]
+        val previousPub = latestPubs[1]
+        val changes = publicationDao.fetchPublicationLocationTrackChanges(latestPub.id)
+
+        val diff = publicationService.diffLocationTrack(
+            localizationService.getLocalization(LocalizationLanguage.FI),
+            changes.getValue(originalLocationTrack.id),
+            publicationDao.fetchPublicationLocationTrackSwitchLinkChanges(latestPub.id)[originalLocationTrack.id],
+            LayoutBranch.main,
+            latestPub.publicationTime,
+            previousPub.publicationTime,
+            trackNumberDao.fetchTrackNumberNames(),
+            setOf(),
+        ) { _, _ -> null }
+        assertEquals(1, diff.size)
+        assertEquals("linked-switches", diff[0].propKey.key.toString())
+        assertEquals(
+            """
+                Vaihteiden sw-deleted, sw-replaced-with-new-same-name (1.1.1.1.8), sw-unlinked-from-alignment,
+                sw-unlinked-from-topology linkitys purettu. Vaihteet sw-added-to-alignment, sw-added-to-topo-end,
+                sw-added-to-topo-start, sw-replaced-with-new-same-name (1.1.1.1.9) linkitetty.
+            """.trimIndent().replace("\n", " "),
```

Comment:
Vähän ikävähköä että näitä joutuu käännöksillä tarkistamaan 🤔

Reply:
Tämä on joo näiden puhtaasti bäkkärillä käännettävien fraasien huono puoli. Toinen vaihtoehto lienisi että kuljetettaisiin jotenkin mukana käännösavainta ja parametreja, mutta toisaalta niitä joutuisi kuljettamaan mukana pelkästään testejä varten.